### PR TITLE
Add FTP estimation from power curve

### PR DESCRIPTION
**Pull request by LassiHeikkila**
## Summary

Estimate FTP from the athlete's power curve using two methods, show both on the Power view, and let the user accept either to set their profile FTP.

- **Simple** — FTP = 95% of the 20-minute (1200s) best power.
- **Critical Power** — linear work–time model fit over the 2–20 minute bests (`W(t) = CP·t + W'`, closed-form OLS, no new dependencies). CP is used directly as the FTP estimate; `CP` and `W'` are surfaced for transparency.

## Backend

- `openkoutsi/training_math.py`: `estimate_ftp_simple`, `estimate_cp_wprime`, `CP_FIT_DURATIONS`.
- New `GET /power/ftp-estimate` endpoint (`FtpEstimateResponse`) with optional `?days` window, using the rank-1 best per duration and availability flags.
- `AthleteUpdate` gains an optional `ftp_test_method`, so accepted estimates are recorded in `ftp_tests` as `"20min"`/`"cp"` (defaults to `"manual"`, backward compatible).

## Frontend

- FTP estimate card on the Power view with **Accept** buttons that PUT the chosen FTP and refresh the athlete app-wide.
- Shows "Not enough data" when bests are missing; CP row shows `CP / W′`.
- EN + FI translations added.

## Tests & docs

- Unit tests for the estimation math (known-value CP/W' recovery, edge cases).
- Integration tests for the endpoint (both/one/neither method available, `?days` filtering) and for recording the FTP test method.
- `openapi.json` and `README.md` updated.

## Verification note

The sandbox blocks pip/npm downloads, so the full `pytest`/`vitest` suites could not be run locally — CI will exercise them. Verified locally: the pure-Python CP/FTP math against known-value cases, `py_compile` on all changed backend/test files, and frontend import/i18n resolution.

https://claude.ai/code/session_01WU6qck83kqxvfjiSFtSJAT

---
_Generated by [Claude Code](https://claude.ai/code/session_01WU6qck83kqxvfjiSFtSJAT)_

### CODE REVIEW

File: backend/app/schemas/athlete.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ class AthleteUpdate(BaseModel):
     hr_zones: Optional[list[ZoneSchema]] = None
     power_zones: Optional[list[ZoneSchema]] = None
     app_settings: Optional[dict] = None
+    ftp_test_method: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (LassiHeikkila) [reply backend/app/schemas/athlete.py]: Good catch — constrained the API input to `Literal["manual", "20min", "cp"]` in 687d98f so unknown values are now rejected with a 422 instead of being persisted verbatim. Added an integration test covering the rejection. (The stored `ftp_tests[*].method` model stays free-form, since other code paths use values like `"test"`; this only validates the user-supplied API input.)

---
_Generated by [Claude Code](https://claude.ai/code)_
